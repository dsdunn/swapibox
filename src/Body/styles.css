.body {
  display: flex;
  margin-left: 250px;
  margin-top: 3%;
  flex-wrap: wrap;
  justify-content: space-around;
  height: 80vh;
  overflow: scroll; }
