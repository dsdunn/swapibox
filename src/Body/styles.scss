.body {
  display: flex;
  margin-left:250px;
  flex-wrap: wrap;
  justify-content: space-around;
  background-color: #1c1c12;
}

