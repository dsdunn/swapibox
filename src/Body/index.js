import React from 'react';
import Card from '../Card';
import PropTypes from 'prop-types';
import './styles.css'

const Body = ({state, toggleFavorites}) => {

  const makeCards = (group) => {
    return group.map(individual => {
      return <Card 
          individual={individual} 
          key={individual.name} 
          toggleFavorites={toggleFavorites} 
          isFavorite={isFavorite(individual)}
        />
    })
  }

  const group = state.category === 'people' ? state.people :
    state.category === 'vehicles' ? state.vehicles :
    state.category === 'planets' ? state.planets : 
    state.category === 'favorites' ? state.favorites : []

  const isFavorite = (card) => {
    return state.favorites.filter(fav => card.name === fav.name).length > 0
  }

  return (
      <div className="body">
        {state.category === 'favorites' && !group.length ?
          <h2>no favorites</h2> :
          makeCards(group)
        }
      </div>
    )
}

Body.propTypes = {
  state: PropTypes.object.isRequired,
  toggleFavorites: PropTypes.func.isRequired
}

export default Body;