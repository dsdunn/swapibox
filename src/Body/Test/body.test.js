import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Body from '../'

describe('Body',  () => {
  let wrapper;
  let mockState = {
    category: 'people',
    people: [{name:'joe'}],
    vehicles: [{class: 'junk'}, {class: 'luxury'}],
    favorites: [],
    filmInfo: {some: "info"}
  }
  const mockToggleFavs = jest.fn();

  it('should match the snapshot', () => {
    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render cards for each item in current category', () => {
    mockState = {
    category: 'people',
    people: [{name:'joe'}],
    vehicles: [{class: 'junk'}, {class: 'luxury'}],
    planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
    favorites: [],
    filmInfo: {some: "info"}
  }

    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

    expect(wrapper.find('Card').length).toEqual(1);

    mockState = {
      category: 'vehicles',
      people: [{name:'joe'}],
      vehicles: [{class: 'junk'}, {class: 'luxury'}],
      planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
      favorites: [],
      filmInfo: {some: "info"}
    }

    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

    expect(wrapper.find('Card').length).toEqual(2);

    mockState = {
      category: 'favorites',
      people: [{name:'joe'}],
      vehicles: [{class:'junk'}, {class:'luxury'}],
      planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
      favorites: [{class:'junk'}, {name:'joe'}, {name:'thatOne'}, {class:'luxury'}],
      filmInfo: {some: "info"}
    }

    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

      
    expect(wrapper.find('Card').length).toEqual(4);

    mockState = {
      category: 'planets',
      people: [{name:'joe'}],
      vehicles: [{class: 'junk'}, {class: 'luxury'}],
      planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
      favorites: [],
      filmInfo: {some: "info"}
    }

    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

    expect(wrapper.find('Card').length).toEqual(3);
  })

  it('should render a noFavorites message if category is favorite and favorites is empty', () => {
    mockState = {
      category: 'favorites',
      people: [{name:'joe'}],
      vehicles: [{class: 'junk'}, {class: 'luxury'}],
      planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
      favorites: [],
      filmInfo: {some: "info"}
    }
    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

    expect(wrapper.find('Cards').length).toEqual(0);
    expect(wrapper.find('h2').length).toEqual(1)

  })

  it('should render no cards if state is empty', () => {
    mockState = {
      category: '',
      people: [{name:'joe'}],
      vehicles: [{class: 'junk'}, {class: 'luxury'}],
      planets: [{name:'thisOne'}, {name:'thatOne'}, {name:'anotherOne'}],
      favorites: [],
      filmInfo: {some: "info"}
    }
    wrapper = shallow(<Body 
                      state={mockState} 
                      toggleFavorites={mockToggleFavs}
                      />)

    expect(wrapper.find('Card').length).toEqual(0);
  })

})