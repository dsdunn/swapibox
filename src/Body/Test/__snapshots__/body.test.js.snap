// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body
    state={
        Object {
            "category": "people",
            "favorites": Array [],
            "filmInfo": Object {
              "some": "info",
            },
            "people": Array [
              Object {
                "name": "joe",
              },
            ],
            "vehicles": Array [
              Object {
                "class": "junk",
              },
              Object {
                "class": "luxury",
              },
            ],
          }
    }
    toggleFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          individual={
                    Object {
                              "name": "joe",
                            }
          }
          isFavorite={false}
          toggleFavorites={[Function]}
/>,
      ],
      "className": "body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "joe",
        "nodeType": "function",
        "props": Object {
          "individual": Object {
            "name": "joe",
          },
          "isFavorite": false,
          "toggleFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            individual={
                        Object {
                                    "name": "joe",
                                  }
            }
            isFavorite={false}
            toggleFavorites={[Function]}
/>,
        ],
        "className": "body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "joe",
          "nodeType": "function",
          "props": Object {
            "individual": Object {
              "name": "joe",
            },
            "isFavorite": false,
            "toggleFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
